app/models/user.rb

```diff
@@ -46,6 +46,10 @@ def account_id
     @account_id ||= account&.id
   end
 
+  def credential_lock
+    @credential_lock ||= user_verification&.locked
```

Comment:
We need this to be like the `needs_accepted_terms_of_use` function, otherwise this won't memoize if this is `false` (basically make sure to make the call if `credential_lock` is `nil`, but don't make it if `credential_lock` is `false` or `true`. Also, add `credential_lock` to the `attribute` definition so it'll be added on the redis model